src/current/v23.1/connection-pooling.md

```diff
@@ -25,37 +25,149 @@ Each time an application reads or writes data, it will request one of the connec
+If possible configure your connection pool to avoid periodic spikes in new connections, also known as "thundering herds" or "connection storms." Most connection pools offer the ability to stagger their connection age through the use of a connection lifetime "jitter," where the wait time between new connections is randomized. Using a connection jitter of 10% of the maximum connection lifetime smooths out the rate of new connections to a cluster. For example, for a maximum connection lifetime of 300 seconds, set the connection jitter to 30 seconds.
 
-Cockroach Labs performed lab testing of various customer workloads and found no improvement in scalability beyond:
+### Validate connections in a pool
+
+After a connection pool initializes connections to CockroachDB clusters, those connections can occasionally break. This could be due to changes in the cluster topography, or rolling upgrades and restarts, or network disruptions.
+
+Validating connections is typically handled automatically by the connection pool. For example, in HikariCP the connection is validated whenever you request a connection from the pool, and the `keepaliveTime` property allows you to configure an interval to periodically check if the connections in the pool are valid. Whatever connection pool you use, make sure connection validation is enabled when running your application.
+
+#### Implement connection retry logic in your application
+
+To be resilient to connection closures, your application should use a retry loop to reissue transactions that were open when a connection was closed. For details, see [Connection retry loop]({% link {{ page.version.version }}/node-shutdown.md %}#connection-retry-loop).
+
+<section class="filter-content" markdown="1" data-scope="dedicated selfhosted">
+
+If you cannot tolerate connection errors during node drain, you can change the `server.shutdown.connection_wait` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) to allow SQL client connections to gracefully close before CockroachDB forcibly closes them. For guidance, see [Node Shutdown]({% link {{ page.version.version }}/node-shutdown.md %}#server-shutdown-connection_wait).
+
+</section>
+
+## Size connection pools
+
+<div class="filters clearfix">
+  <button class="filter-button page-level" data-scope="serverless"><strong>{{ site.data.products.serverless }}</strong></button>
+  <button class="filter-button page-level" data-scope="dedicated"><strong>{{ site.data.products.dedicated }}</strong></button>
+  <button class="filter-button page-level" data-scope="selfhosted"><strong>{{ site.data.products.core }}</strong></button>
+</div>
+
+If your connection pool is properly configured, the total number of connections to your cluster should be at least 100 times larger than the number of new connections per second to your cluster.
```

Comment:
I'd soften this advice a bit, just b/c there may be valid cases where this advice does not apply. Maybe throw in the word "typical" or "usually", etc.